### PR TITLE
Add SSL/TLS support to nginx input plugin

### PR DESCRIPTION
**Pull request by bobmshannon**
This PR adds support for an optional SSL/TLS config in the nginx input plugin.
 
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by danielnelson** [comment]
Thanks the the PR, we need to make sure we don't create a new transport per interval.  I recently fixed this in a few other plugin in #2782.

**Comment by bobmshannon** [comment]
Hey @danielnelson, thanks for the feedback. I've updated this PR so that the transport is re-used, using the apache input plugin as an example for doing so.

File: plugins/inputs/nginx/nginx.go [hunk 1]
```diff
@@ -12,16 +12,41 @@ import (
...
+	client *http.Client
+	// Response timeout
+	ResponseTimeout internal.Duration
 }
 
 var sampleConfig = `
-  ## An array of Nginx stub_status URI to gather stats.
-  urls = ["http://localhost/status"]
+  # Read Nginx's basic status information (ngx_http_stub_status_module)
+  [[inputs.nginx]]
```
Comment (danielnelson) [inline plugins/inputs/nginx/nginx.go]: Remove the toml table header here, it is added automatically when outputting the sample config.

File: plugins/inputs/nginx/nginx.go [hunk 2]
```diff
@@ -12,16 +12,41 @@ import (
...
+    # An array of Nginx stub_status URI to gather stats.
+    urls = ["http://localhost/server_status"]
+
+    # TLS/SSL configuration
+    ssl_ca = "var/security/ca.pem"
+    ssl_cert = "var/security/cert.cer"
+    ssl_key = "var/security/key.key"
```
Comment (danielnelson) [inline plugins/inputs/nginx/nginx.go]: Change the paths to `/etc/telegraf/<file>` to match other plugins.

File: plugins/inputs/nginx/nginx.go [hunk 3]
```diff
@@ -12,16 +12,41 @@ import (
...
+    insecure_skip_verify = false
+
+    # HTTP response timeout (default: 5s)
+    response_timeout = "10s"
```
Comment (danielnelson) [inline plugins/inputs/nginx/nginx.go]: Set this to 5s since that's the default

**Comment by danielnelson** [comment]
Thank you!  This will be in 1.4.

**Pull request merged**